### PR TITLE
feat: add sms template

### PR DESCRIPTION
**Pull request by israx**
*Issue #, if available:*

*Description of changes:*

Add OTP SMS template

**DX:**
```ts
defineAuth({
  loginWith: {
    email: true,
  },
  passwordlessOptions: {
    otp: {
      userVerification: {
        smsMessage: 'My custom sms message with verification code',
      },
    },
  },
});

```
**Other changes:**
- avoids creating env variables if `originationNumber`, `senderId` or `otpLength` are not defined

**TODOs:**
- add Email template for OTP case
- add Email template for magic link case

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/passwordless-auth-construct/src/otp/construct.ts [hunk 1]
```diff
@@ -43,10 +43,13 @@ export class AmplifyOtpAuth extends Construct {
...
       senderId: props.senderId,
       otpLength: props.length?.toString(),
+      smsMessage: props.userVerification?.smsMessage,
+      emailSubject: props.userVerification?.emailSubject,
+      emailBody: props.userVerification?.emailBody,
     };
 
     for (const [key, value] of Object.entries(createAuthChallengeEnvVars)) {
-      triggers.createAuthChallenge.addEnvironment(key, value ?? '');
+      value && triggers.createAuthChallenge.addEnvironment(key, value);
```
Comment (israx) [inline packages/passwordless-auth-construct/src/otp/construct.ts:52]: avoids creating env variables if values are undefined

File: packages/passwordless-auth-construct/API.md [hunk 2]
```diff
@@ -30,6 +31,10 @@ export type PasswordlessAuthProps = {
     otp?: OtpAuthOptions | boolean;
 };
 
+// Warnings were encountered during analysis:
+//
+// src/types.ts:181:3 - (ae-forgotten-export) The symbol "OtpUserVerification" needs to be exported by the entry point index.d.ts
+
```
Comment (Equartey) [inline packages/passwordless-auth-construct/API.md:37]: blocking: This error needs addressing

File: packages/passwordless-auth-construct/src/types.ts [hunk 3]
```diff
@@ -123,6 +123,32 @@ export type MagicLinkAuthOptions = {
...
+ * Options for SMS user verification
+ */
+export type SmsUserVerificationOptions = {
+  smsMessage?: string;
+};
+
+/**
+ * Options for OTP user verification
+ */
+export type OtpUserVerification = EmailUserVerificationOptions &
```
Comment (Equartey) [inline packages/passwordless-auth-construct/src/types.ts:144]: nit: Don't think `UserVerification` fits well here. Maybe `MessageOptions`? Doesn't need to be changed now, I think these will get ironed out in API review.

File: packages/backend-auth/API.md [hunk 4]
```diff
@@ -28,6 +29,7 @@ export type AmazonProviderFactoryProps = Omit<AmazonProviderProps, 'clientId' |
 // @public (undocumented)
 export type AmplifyAuthFactoryProps = Omit<AuthProps, 'outputStorageStrategy' | 'loginWith'> & TriggerConfig & {
     loginWith: AuthLoginWithFactoryProps;
+    passwordlessOptions?: PasswordlessAuthProps;
```
Comment (sobolk) [inline packages/backend-auth/API.md:32]:
```suggestion
    passwordless?: PasswordlessAuthProps;

```

File: packages/passwordless-auth-construct/API.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+```ts
+
+import { AmplifyAuth } from '@aws-amplify/auth-construct-alpha';
+import { Construct } from 'constructs';
+
+// @public
+export class AmplifyPasswordlessAuth extends Construct {
+    constructor(scope: Construct, id: string, auth: AmplifyAuth, props: PasswordlessAuthProps);
+}
```
Comment (sobolk) [inline packages/passwordless-auth-construct/API.md:13]: Something is odd with this PR.
The `passwordless-auth-construct` was added to feature branch already.
see https://github.com/aws-amplify/amplify-backend/commits/feat/passwordless-auth/packages/passwordless-auth-construct/API.md .

Are you using correct base branch for this PR locally?

File: packages/backend-auth/API.md [hunk 6]
```diff
@@ -84,6 +90,11 @@ export type OidcProviderFactoryProps = Omit<OidcProviderProps, 'clientId' | 'cli
     clientSecret: BackendSecret;
 };
 
+// @public (undocumented)
+export type TriggerConfig = {
+    triggers?: Partial<Record<TriggerEvent, ConstructFactory<ResourceProvider<FunctionResources>>>>;
+};
```
Comment (sobolk) [inline packages/backend-auth/API.md:96]: This already exists on main here https://github.com/aws-amplify/amplify-backend/blob/7398baa80461dee7d3abd47b51926c640fb6b6e8/packages/backend-auth/src/factory.ts#L26-L36 .

File: packages/backend-auth/API.md [hunk 7]
```diff
@@ -26,6 +26,12 @@ export type AmazonProviderFactoryProps = Omit<AmazonProviderProps, 'clientId' |
     clientSecret: BackendSecret;
 };
 
+// @public (undocumented)
+export type AmplifyAuthFactoryProps = Omit<AuthProps, 'outputStorageStrategy' | 'loginWith'> & TriggerConfig & {
```
Comment (sobolk) [inline packages/backend-auth/API.md:30]: this type already exists here https://github.com/aws-amplify/amplify-backend/blob/7398baa80461dee7d3abd47b51926c640fb6b6e8/packages/backend-auth/src/factory.ts#L20-L21

**Pull request closed without merging**